### PR TITLE
use checked write in `LineWriter` example

### LINKED ISSUE #51621: Docs: unchecked file.write() in example
In the [example for LineWriter](https://doc.rust-lang.org/std/io/struct.LineWriter.html) there is this code fragment:
```
    for &byte in road_not_taken.iter() {
       file.write(&[byte]).unwrap();
    }
```
This code isn't correct -- it should be `file.write_all` instead. In fact, trying the run the example fails on `play.rust-lang.org`.

### PR DESCRIPTION
**Pull request by euclio**
The example was wrong because it didn't check the return value of
`write()`, and it didn't flush the buffer before comparing the contents
of the file.

Fixes #51621.

### CODE REVIEW

File: src/libstd/io/buffered.rs [hunk 1]
```diff
@@ -752,11 +752,13 @@ impl<W> fmt::Display for IntoInnerError<W> {
 ///     let file = File::create("poem.txt")?;
 ///     let mut file = LineWriter::new(file);
 ///
-///     for &byte in road_not_taken.iter() {
-///        file.write(&[byte]).unwrap();
-///     }
+///     file.write_all(road_not_taken)?;
```
Comment (sfackler) [inline src/libstd/io/buffered.rs]: The example is explicitly writing each byte individually to note how the buffering provided by LineWriter helps things. If we were going to write the whole message out at once, there's no reason to use the LineWriter at all.

Comment (euclio) [reply src/libstd/io/buffered.rs]: It's still line buffering behind the scenes, so I don't think that it's totally wrong. I'm certainly open to suggestions on how to improve the entire example, though. Maybe we could write some bytes, and demonstrate that they aren't written to disk until a newline is written?

Comment (sfackler) [reply src/libstd/io/buffered.rs]: That seems reasonable.

**Comment by TimNN** [comment]
Ping from triage, @withoutboats : This PR needs your review!

**Comment by euclio** [comment]
This isn't ready for review, I plan to rewrite the example per sfackler's review.

**Comment by euclio** [comment]
@sfackler @withoutboats This is ready for another review.

File: src/libstd/io/buffered.rs [hunk 2]
```diff
@@ -752,17 +752,31 @@ impl<W> fmt::Display for IntoInnerError<W> {
...
 ///     let mut file = LineWriter::new(file);
 ///
-///     for &byte in road_not_taken.iter() {
-///        file.write(&[byte]).unwrap();
-///     }
+///     file.write_all(b"I shall be telling this with a sigh")?;
+///
+///     // No bytes are written until a newline is encountered (or
+///     // the internal buffer is filled).
+///     assert_eq!(fs::read_to_string("poem.txt")?.as_bytes(), b"");
```
Comment (frewsxcv) [inline src/libstd/io/buffered.rs]: I think `fs::read_to_string(...)?.as_bytes()` can just be `fs::read(...)?`

https://doc.rust-lang.org/nightly/std/fs/fn.read.html

Comment (sfackler) [reply src/libstd/io/buffered.rs]: Alternatively this can be `assert_eq!(fs::read_to_string("poem.txt")?, "")`.

Comment (euclio) [reply src/libstd/io/buffered.rs]: Fixed.

**Comment by TimNN** [comment]
Ping from triage, @withoutboats: This PR is waiting for your review.

**Review by frewsxcv — APPROVED** [review]
This looks great, thanks for this 🎉

**Pull request merged**